apollo-ios/Sources/Apollo/FieldPolicyDirectiveEvaluator.swift

```diff
@@ -0,0 +1,234 @@
+import Foundation
+#if !COCOAPODS
+import ApolloAPI
+#endif
+
+enum FieldPolicyResult {
+  case single(CacheKeyInfo)
+  case list([CacheKeyInfo])
+}
+
+struct FieldPolicyDirectiveEvaluator {
```

Comment:
This is some really nice code. Lots of complex logic and edge cases to deal with. Great job here.